### PR TITLE
DINT-77: Add dashboard widget definition

### PR DESCRIPTION
**Pull request by shivtej1505**

### CODE REVIEW

**Comment by shivtej1505** [comment]
@ssmusoke , here are the changed I made in data integrity module.

File: omod/src/main/resources/apps/dataintegrity_app.json [hunk 1]
```diff
@@ -12,5 +12,28 @@
         "order": 300
       }
     ]
+  },
+  {
+    "id": "ugandaemr.dataIntegrityViolations",
```
Comment (ssmusoke) [inline omod/src/main/resources/apps/dataintegrity_app.json]: Rename id to coreapps.dataIntegrityViolations

File: omod/src/main/resources/apps/dataintegrity_app.json [hunk 2]
```diff
@@ -12,5 +12,28 @@
...
+    "config": {
+      "widget": "dataintegrityviolations",
+      "icon": "icon-question-sign",
+      "label": "Data Quality Violations",
+      "maxResults": "8"
+    },
+    "extensions": [
+      {
+        "id": "org.openmrs.module.coreapps.mostRecentVitals.clinicianDashboardFirstColumn",
+        "appId": "ugandaemr.dataIntegrityViolations",
```
Comment (shivtej1505) [inline omod/src/main/resources/apps/dataintegrity_app.json]: @ssmusoke , I've made the change you're suggested but now I'm getting new error. Here is full log: https://pastebin.com/V1xXyy1L

After getting the error I further changed the `appId` to `coreapps.dataIntegrityViolations` then I got no error but I wasn't able to see the content of `data quality widget`.

Comment (ssmusoke) [reply omod/src/main/resources/apps/dataintegrity_app.json]: You did not change the appId in the extensions section

**Comment by shivtej1505** [comment]
@ssmusoke , updated the appId. Here is the screenshot:
screen shot 2017-06-22 at 8 35 13 pm

Also following resource are not being loaded, if may be the problem:

* http://localhost:8080/openmrs/ms/uiframework/resource/coreapps/scripts/web/coreapps.dashboardwidgets.js
* http://localhost:8080/openmrs/ms/uiframework/resource/coreapps/scripts/web/coreapps.vendor.js
* http://localhost:8080/openmrs/ms/uiframework/resource/uicommons/scripts/datetimepicker/locales/bootstrap-datetimepicker.en_GB.js

**Comment by shivtej1505** [comment]
@ssmusoke , found this link: https://talk.openmrs.org/t/solved-qa-refapp-openmrs-org-does-not-have-the-latest-version-of-coreapps-module/11661 . Trying it.

**Comment by shivtej1505** [comment]
@ssmusoke , I found the problem. Somehow `coreapps.vendor.js` and `coreapps.dashboardwidgets.js` are not generated. I copied them from the `qa-refapp.openmrs.org` and the all the widgets worked. 

I'm trying to get familiarise with the webpack so I can solve this problem, if you have any solution please let me know.

**Pull request merged**